src/flat/faabric.fbs

```diff
@@ -1,24 +1,33 @@
+table SnapshotMergeRegionRequest {
+  offset:int;
+  length:ulong;
+  data_type:int;
+  merge_op:int;
+}
+
 table SnapshotPushRequest {
   key:string;
-  maxSize:ulong;
+  max_size:ulong;
   contents:[ubyte];
+  merge_regions:[SnapshotMergeRegionRequest];
 }
 
 table SnapshotDeleteRequest {
   key:string;
 }
 
-table SnapshotDiffChunk {
+table SnapshotDiffRequest {
   offset:int;
-  dataType:int;
-  mergeOp:int;
+  data_type:int;
+  merge_op:int;
```

Comment:
`SnapshotDiff` and `SnapshotMergeRegion` names clash with those in `faabric::util`. I've appended `Request` as this fits with the style of the rest of the FB objects. 

I also switched these fields to all use snake case as FB prints a warning if you do anything else.